### PR TITLE
Fixes multi-z photography

### LINKED ISSUE #48992: Cameras see past the illusion of openspace
image

image

very illegal

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

Fixes #48992 

**BEFORE:**
image

**AFTER:**
image

Note to self: Stop coding while running on no sleep and a lot of caffeine.

## Why It's Good For The Game

Now you can stalk people from above, and also I can claim one of the game keys @MMMiracles is offering for fixing multi-z bugs.

## Changelog
:cl: ike709
fix: Taking pictures of openspace turfs works now.
/:cl:

### CODE REVIEW

File: code/modules/photography/camera/camera.dm [hunk 1]
```diff
@@ -180,8 +180,17 @@
...
 	var/clone_area = SSmapping.RequestBlockReservation(size_x * 2 + 1, size_y * 2 + 1)
-	for(var/turf/T in block(locate(target_turf.x - size_x, target_turf.y - size_y, target_turf.z), locate(target_turf.x + size_x, target_turf.y + size_y, target_turf.z)))
-		if((ai_user && GLOB.cameranet.checkTurfVis(T)) || (T in seen))
+	for(var/turf/placeholder in block(locate(target_turf.x - size_x, target_turf.y - size_y, target_turf.z), locate(target_turf.x + size_x, target_turf.y + size_y, target_turf.z)))
+		var/turf/T = placeholder
+
+		var/turf_is_below = FALSE
+		while(istype(T, /turf/open/openspace)) //Multi-z photography
+			T = SSmapping.get_turf_below(T)
+			if(!T)
```
Comment (ike709) [inline code/modules/photography/camera/camera.dm]: This is probably unnecessary, but better safe than sorry I guess?

**Comment by Dennok** [comment]
This nice but camera see past darkness.

**Comment by ike709** [comment]
@Dennok Cameras intentionally don't capture darkness, they have a flash when you use them.

File: code/modules/photography/camera/camera.dm [hunk 2]
```diff
@@ -180,8 +180,17 @@
...
+				break
+			turf_is_below = TRUE
+		
+		if(T && ((ai_user && GLOB.cameranet.checkTurfVis(T)) || (T in seen) || turf_is_below))
```
Comment (Cyberboss) [inline code/modules/photography/camera/camera.dm]:
```suggestion
		if(T && ((ai_user && GLOB.cameranet.checkTurfVis(placeholder)) || (placeholder in seen))
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**